guarddog/analyzer/sourcecode/npm-silent-process-execution.yml

```diff
@@ -0,0 +1,46 @@
+rules:
+- id: npm-silent-process-execution
+  message: silent process execution
+  patterns:
+    - pattern-either:
+      - pattern-inside: |
```

Comment:
can we also add cases for ESM and domained imports?
```
import CP from 'node:child_process';
import { execSync } from 'node:child_process';
import CP from 'child_process';
import { execSync } from 'child_process';
require('node:child_process')
```

Reply:
Added - didn't add any unit tests as it's pointless (we'd need 1 file per use-case to properly test it). Also asked the Semgrep team how this could be better, because it's a mess

https://semgrep.slack.com/archives/C018NJRRCJ0/p1672916228577459